### ISSUE TITLE
opencv=4.7.0 uninstallable with GDAL 3.6.4, HDF5 1.14.0 and Python 3.11

### ISSUE THREAD

**Issue by weiji14** [conda-forge/opencv-feedstock#357]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Trying to solve for an Python 3.11 recently and am getting some incompatibilities. Not sure if this is ther correct feedstock, but I've narrowed it down to some quirk with `opencv` and HDF5 (possibly related to #351?). Here's a minimal `environment.yml` to reproduce:

```
name: temp
channels:
  - conda-forge
  - nodefaults
dependencies:
  - gdal>=3.6.4
  - hdf5>=1.14.0
  - opencv>=4.7.0
  - python>=3.11
```

Running `mamba env create -f environment.yml` gives this traceback:

```bash
conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache


Looking for: ["gdal[version='>=3.6.4']", "hdf5[version='>=1.14.0']", "opencv[version='>=4.7.0']", "python[version='>=3.11']"]


Could not solve for environment specs
The following packages are incompatible
├─ gdal >=3.6.4  is installable with the potential options
│  ├─ gdal 3.6.4 would require
│  │  └─ libgdal 3.6.4 h7239d12_0, which requires
│  │     ├─ libcurl >=8.0.1,<9.0a0 , which requires
│  │     │  └─ openssl >=3.1.0,<4.0a0 , which can be installed;
│  │     └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which requires
│  │        └─ jpeg <0.0.0a , which can be installed;
│  ├─ gdal 3.6.4 would require
│  │  ├─ hdf5 >=1.14.0,<1.14.1.0a0  with the potential options
│  │  │  ├─ hdf5 1.14.0, which can be installed;
│  │  │  └─ hdf5 1.14.0 would require
│  │  │     └─ openssl >=1.1.1t,<1.1.2a , which conflicts with any installable versions previously reported;
│  │  ├─ libgdal 3.6.4 h7239d12_1, which requires
│  │  │  ├─ hdf5 >=1.14.0,<1.14.1.0a0  with the potential options
│  │  │  │  ├─ hdf5 1.14.0, which can be installed;
│  │  │  │  └─ hdf5 1.14.0, which cannot be installed (as previously explained);
│  │  │  ├─ libcurl >=8.0.1,<9.0a0 , which can be installed (as previously explained);
│  │  │  └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which can be installed (as previously explained);
│  │  └─ python_abi 3.10.* *_cp310, which can be installed;
│  ├─ gdal 3.6.4 would require
│  │  ├─ libgdal 3.6.4 h7239d12_1, which can be installed (as previously explained);
│  │  └─ python >=3.11,<3.12.0a0 , which requires
│  │     └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
│  ├─ gdal 3.6.4 would require
│  │  └─ libgdal 3.6.4 h7239d12_1, which can be installed (as previously explained);
│  └─ gdal 3.6.4 would require
│     ├─ hdf5 >=1.14.0,<1.14.1.0a0  with the potential options
│     │  ├─ hdf5 1.14.0, which can be installed;
│     │  └─ hdf5 1.14.0, which cannot be installed (as previously explained);
│     ├─ libgdal 3.6.4 h7239d12_1, which can be installed (as previously explained);
│     └─ python_abi 3.9.* *_cp39, which can be installed;
├─ hdf5 >=1.14.0  is installable with the potential options
│  ├─ hdf5 1.14.0, which can be installed;
│  └─ hdf5 1.14.0, which cannot be installed (as previously explained);
├─ opencv >=4.7.0  is installable with the potential options
│  ├─ opencv 4.7.0 would require
│  │  └─ libopencv [4.7.0 py310h5bd1119_0|4.7.0 py310hb48cf42_1|...|4.7.0 py39hb375605_0], which requires
│  │     └─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
│  ├─ opencv 4.7.0 would require
│  │  └─ py-opencv 4.7.0 py310hfdc917e_2, which requires
│  │     └─ python_abi 3.10.* *_cp310, which can be installed;
│  ├─ opencv 4.7.0 would require
│  │  └─ libopencv [4.7.0 py311hce3c4d3_2|4.7.0 py38h4a38063_2|4.7.0 py38h9896f59_2|4.7.0 py39h0293f97_2], which requires
│  │     └─ hdf5 >=1.12.2,<1.12.3.0a0 , which conflicts with any installable versions previously reported;
│  └─ opencv 4.7.0 would require
│     └─ py-opencv 4.7.0 py39hcca971b_2, which requires
│        └─ python_abi 3.9.* *_cp39, which can be installed;
└─ python >=3.11 , which cannot be installed (as previously explained).
```

I had a look at #354 which seems to have been built with HDF5 1.14.0? But not sure why setting Python 3.11 seems to cause the solver to choose `hdf5=1.12.2`. Could be something on [OpenSSL](https://github.com/conda-forge/openssl-feedstock) also? May need to do some rerendering or something.

### Installed packages

```shell
None
```

### Environment info

```shell
active environment : base
    active env location : /home/user/mambaforge
            shell level : 1
       user config file : /home/user/.condarc
 populated config files : /home/user/mambaforge/.condarc
          conda version : 22.9.0
    conda-build version : not installed
         python version : 3.10.8.final.0
       virtual packages : __cuda=12.0=0
                          __linux=5.19.0=0
                          __glibc=2.35=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/user/mambaforge  (writable)
      conda av data dir : /home/user/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/user/mambaforge/pkgs
                          /home/user/.conda/pkgs
       envs directories : /home/user/mambaforge/envs
                          /home/user/.conda/envs
               platform : linux-64
             user-agent : conda/22.9.0 requests/2.28.1 CPython/3.10.8 Linux/5.19.0-42-generic ubuntu/22.04.2 glibc/2.35
                UID:GID : 1001:1001
             netrc file : /home/user/.netrc
           offline mode : False
```

**Comment by weiji14** [comment]
Dependency conflict appears to have been resolved in #360 which deleted the `recipe/conda_build_config.yaml` file such that only `hdf5=1.14` builds are done now (no more `hdf5=1.12`?). Closing as done.

**Comment by hmaarrfk** [comment]
Not sure why that resolved things but I'm glad it is resolved for you.

Correct, no more 1.12
